### PR TITLE
Update and cleanup Celery integration

### PR DESCRIPTION
**Pull request by antonpirker**
Cleanup the Celery integration to make it easier to work on it and also update to the new minimum supported Celery versions and the new Scopes API.

The easiest way to review this is probably looking at the "Fixed trace propagation" commit:
https://github.com/getsentry/sentry-python/pull/2891/commits/a723cb7b99eca1024bff6b7a5b247ac0904ea65a

The other commits are mostly moving code around.

### CODE REVIEW

**Review by szokeasaurusrex — COMMENTED** [review]
I read through your changes, but I am having trouble understanding the PR at a high level, so I think it would be better if @sentrivana also reviews this.

File: sentry_sdk/integrations/celery/beat.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    if "headers" in headers:
+        headers.update(headers["headers"])
+        del headers["headers"]
+
+    headers.update(task.request.get("properties") or {})
+
+    return headers
+
+
+def _get_humanized_interval(seconds):
```
Comment (sentrivana) [inline sentry_sdk/integrations/celery/beat.py:57]: We could move this to `celery.utils`

**Review by sentrivana — APPROVED** [review]
Looks much better! 🥔 Left a few comments.

File: sentry_sdk/integrations/celery/beat.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/integrations/celery/beat.py]: done

File: sentry_sdk/integrations/celery/__init__.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+
+                    # Add start timestamp of task,
+                    # if Sentry Crons monitoring for Celery Beat tasks is enabled
+                    if integration.monitor_beat_tasks:
+                        headers.update(
+                            {
+                                "sentry-monitor-start-timestamp-s": "%.9f"
+                                % _now_seconds_since_epoch(),
+                            }
+                        )
```
[the remark this answers is not in the document]
Comment (antonpirker) [reply sentry_sdk/integrations/celery/__init__.py]: True! Will test this in the sample apple again to make sure everything behaves like expected!

[the remark this answers is not in the document]
Comment (antonpirker) [reply sentry_sdk/integrations/celery/__init__.py]: Will test this on Thurday

[the remark this answers is not in the document]
Comment (antonpirker) [reply sentry_sdk/integrations/celery/__init__.py]: Ok, tested everything with the current state and looks good!

**Review by sentrivana — APPROVED** [review]

**Pull request merged**